app/frontend/src/features/communities/CommunityPage/DiscussionsSection.tsx

```diff
@@ -113,46 +99,45 @@ export default function DiscussionsSection({
             </Alert>
           )}
           <NewComment
-            onComment={async (content) =>
+            onComment={async (content) => {
               newDiscussionMutation.mutate({
                 content,
                 ownerCommunityId: community.communityId,
                 title: "test",
-              })
-            }
+              });
+            }}
           />
         </DialogContent>
       </Dialog>
       <div className={classes.discussionsContainer}>
         {isDiscussionsLoading && <CircularProgress />}
-        {discussions &&
-          (discussions.pages.length > 0 &&
-          discussions.pages[0].discussionsList.length === 0 ? (
-            <TextBody>{DISCUSSIONS_EMPTY_STATE}</TextBody>
-          ) : (
-            discussions.pages
-              .flatMap((res) => res.discussionsList)
-              .map((discussion) => (
-                <DiscussionCard
-                  discussion={discussion}
-                  className={classes.discussionCard}
-                  key={`discussioncard-${discussion.threadId}`}
-                />
-              ))
-          ))}
+        {hasAtLeastOnePage(discussions, "discussionsList") ? (
+          discussions.pages
+            .flatMap((res) => res.discussionsList)
+            .map((discussion) => (
+              <DiscussionCard
+                discussion={discussion}
+                key={`discussioncard-${discussion.threadId}`}
+              />
+            ))
+        ) : (
+          <TextBody>{DISCUSSIONS_EMPTY_STATE}</TextBody>
+        )}
         {discussionsHasNextPage && (
           <div className={classes.loadMoreButton}>
-            <Link
```

Comment:
Apparently nesting `<button>`s inside `<a>`s (what `Link` from React router generates) is [invalid HTML](https://stackoverflow.com/questions/6393827/can-i-nest-a-button-element-inside-an-a-using-html5#:~:text=No%2C%20it%20isn't%20valid,be%20no%20interactive%20content%20descendant.), so that's why I went with button to keep the styling but use the history API to do the linking.

Reply:
How about `<Link component={Button}>`

Reply:
That caused a new warning saying `navigate` from React Router gets passed to `<a>`, so the button workaround might be the best...

Reply:
Also it is button that just happens to have the same functionality as a link so it's not exactly semantically wrong to use a button...

Reply:
Still not sure about this. Doesn't it break some functionality, like middle mouse click opening in new tab?

How about this - replacing the outer div with `<Button component="div">` and keeping the link? That should be correct html and semantics and still be a proper link

Reply:
> Doesn't it break some functionality, like middle mouse click opening in new tab?

Whether this is "breaking"/missing some functionality depends on if you know the underlying thing is an actual link or not. If you are a new user, you probably don't as it looks like a button?

This also probably crosses the classic debate of "should we have links that look like buttons?". Personally not a fan of mixing the two as I think it's confusing UX, since you don't always know what you're gonna get when you see a button.